contract/models/contract_line.py

```diff
@@ -611,12 +611,19 @@ def _insert_markers(self, first_date_invoiced, last_date_invoiced):
         return name
 
     def _update_recurring_next_date(self):
-        # FIXME: Change method name according to real updated field
-        # e.g.: _update_last_date_invoiced()
         for rec in self:
             last_date_invoiced = rec.next_period_date_end
+            recurring_next_date = rec.get_next_invoice_date(
+                last_date_invoiced + relativedelta(days=1),
+                rec.recurring_invoicing_type,
+                rec.recurring_invoicing_offset,
+                rec.recurring_rule_type,
+                rec.recurring_interval,
+                max_date_end=rec.date_end,
+            )
             rec.write(
                 {
+                    "recurring_next_date": recurring_next_date,
```

Comment:
IMHO This should never happen as the field is computed...

It's quite insane...

Reply:
You're right. It should be `update`, not `write`, but the aim is to update such data.

Reply:
This is only called here, when preparing the next invoice:

https://github.com/OCA/contract/blob/15209ddf952625534d6464671d9cf8da785cfd6f/contract/models/contract.py#L529

so no computed method involved, and I consider it correct.

Reply:
That's exactly what I wanted to say.

This is a method called outside of computes.

Reply:
> Remember that this uses `compute` as onchanges through the computed writable fields, so it's not a traditional compute field

In any case, the field should only be updated through its compute method

Reply:
@rousseldenis the switch to `compute` was for removing onchanges, but they are traditional regular fields with such trigger for updating the field in certain cases, but that doesn't prevent that they can update it via other means, and these ways are when `depends` syntax can't reflect this change, like the one when creating the invoices. Do you understand?

Reply:
> In any case, the field should only be updated through its compute method

We are crossing our answers, but I have replied to this on the previous one. Please check it and tell me any doubt.

Reply:
Then I'm afraid you are not understanding what a computed writable field is. The compute method is just auxiliary, and replace the onchanges. You can (and in a lot of cases, you must) update the fields directly in other methods.

Reply:
Do you have a pointer where Odoo officially explain that ?

I have only this link : https://www.odoo.com/documentation/14.0/fr/developer/reference/addons/orm.html#reference-fields-compute

Reply:
It's explained here: https://youtu.be/OMZbbWrSsfw?t=1158

I'm not sure such date is updated through the `depends` chain, but we can try adding specific tests for such cases. Anyways, we can put update and the update is delayed until the end, avoiding double writing if you prefer.

Reply:
> It's explained here: https://youtu.be/OMZbbWrSsfw?t=1158

As skipped v13, missed the presentation :sweat_smile:  Ok, I'll try to do a PR to update documentation about that.

Reply:
@pedrobaeza I think we have all the needed chain to let the computation works by itself.

The entry point to change it should be the last_invoice_date IMHO.

As for example here if we want to do it manually : https://github.com/OCA/contract/pull/810